consensus/common/src/main/java/tech/pegasys/pantheon/consensus/common/VoteTallyUpdater.java

```diff
@@ -70,18 +66,12 @@ public VoteTally buildVoteTallyFromBlockchain(final Blockchain blockchain) {
    * @param header the header of the block being added
    * @param voteTally the vote tally to update
    */
-  @Override
   public void updateForBlock(final BlockHeader header, final VoteTally voteTally) {
-    final Address candidate = header.getCoinbase();
     if (epochManager.isEpochBlock(header.getNumber())) {
       voteTally.discardOutstandingVotes();
       return;
     }
-
-    if (!candidate.equals(NO_VOTE_SUBJECT)) {
-      final IbftExtraData ibftExtraData = IbftExtraData.decode(header.getExtraData());
-      final Address proposer = IbftBlockHashing.recoverProposerAddress(header, ibftExtraData);
-      voteTally.addVote(proposer, candidate, VoteType.fromNonce(header.getNonce()).get());
-    }
+    final Optional<CastVote> vote = serialiserDeserialiser.extractVoteFromHeader(header);
+    vote.ifPresent(v -> voteTally.addVote(v.getProposer(), v.getRecipient(), v.getVotePolarity()));
```

Comment:
It would be really nice of `VoteTally.addVote` took a `CastVote` instance.  Then you could just write:
`serialiserDeserialiser.extractVoteFromHeader(header).ifPresent(voteTally::addVote)`

Reply:
That's why we get you to review.
done.